### PR TITLE
Stan/typescript migration

### PR DESCRIPTION
**Pull request by stan29308**
Changed name of attendance data mapper and added attendance unit test.

### CODE REVIEW

File: api/lib/models/attendance/attendance-data-mapper-impl.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+export const TABLE_NAME = 'ATTENDANCE';
+
+/**
+ * TODO: Add documentation
+ */
+
+@Injectable()
+export class AttendanceDataMapperImpl extends GenericDataMapper
+  implements IAclPerm {
```
Comment (sushshring) [inline api/lib/models/attendance/attendance-data-mapper-impl.ts]: implements IAttendanceDataMapper

File: api/lib/models/attendance/attendance.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export const TABLE_NAME = 'ATTENDANCE';
+
+/**
+ * TODO: Add documentation
+ */
+export class Attendance extends BaseObject {
+  public idSCANS: number;
+  public scan_location: number;
+  public scan_time: EpochNumber;
+  public user_uid: string;
```
Comment (sushshring) [inline api/lib/models/attendance/attendance.ts]: Change to UidType

File: api/lib/models/attendance/attendance.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  public location_name: string;
+  public event_start_time: EpochNumber;
+  public event_end_time: EpochNumber;
+  public event_title: string;
+  public event_description: string;
+  public event_type: EventType;
+  public hackathon_id: string;
```
Comment (sushshring) [inline api/lib/models/attendance/attendance.ts]: Change to UidType

File: api/lib/models/attendance/attendance.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  public hackathon_name: string;
+  public hackathon_start_time: EpochNumber;
+  public hackathon_end_time: EpochNumber;
+  public hackathon_base_pin: number;
+  public hackathon_active: number;
```
Comment (sushshring) [inline api/lib/models/attendance/attendance.ts]: this should be a boolean

File: api/test/models/attendance/attendence-data-mapper-impl.test.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      endTime: null,
+      name: 'test hackathon',
+      uid: 'test uid',
+    })));
+    activeHackathonDataMapper.tableName.returns('HACKATHON');
+    // Configure Attendance Data Mapper
+    attendanceDataMapper = new AttendanceDataMapperImpl(
+      acl,
+      mysqlUow,
+      activeHackathonDataMapper,
```
Comment (sushshring) [inline api/test/models/attendance/attendence-data-mapper-impl.test.ts]: Does attendaceDataMapper need activeHackathon dependence?

Comment (stan29308) [reply api/test/models/attendance/attendence-data-mapper-impl.test.ts]: Yeah getAll and getCount for the AttendanceDataMapperImpl both use it.

**Pull request merged**